rita/src/rita_common/tunnel_manager/mod.rs

```diff
@@ -38,33 +44,75 @@ type Resolver = Mocker<resolver::Resolver>;
 #[cfg(not(test))]
 type Resolver = resolver::Resolver;
 
-#[derive(Debug, Fail)]
-pub enum TunnelManagerError {
-    #[fail(display = "DNS lookup error")]
-    DNSLookupError,
+/* Uncomment when tunnel state handling is added
+#[derive(Debug, Clone)]
+pub enum TunnelState {
+    Init,
+    Open,
+    Throttled,
+    Closed,
 }
+*/
 
 #[derive(Debug, Clone)]
-pub struct TunnelData {
+pub struct Tunnel {
+    pub ip: IpAddr,
     pub iface_name: String,
+    pub listen_ifidx: u32,
     pub listen_port: u16,
+    //    pub tunnel_state: TunnelState,
+    pub localid: LocalIdentity,
 }
 
-impl TunnelData {
-    fn new(listen_port: u16) -> TunnelData {
+impl Tunnel {
+    fn new(
+        ip: IpAddr,
+        our_listen_port: u16,
+        ifidx: u32,
+        their_id: LocalIdentity,
+    ) -> Result<Tunnel, Error> {
         let iface_name = KI.setup_wg_if().unwrap();
-        TunnelData {
-            iface_name,
-            listen_port,
-        }
+
+        //let init = TunnelState::Init;
+        let tunnel = Tunnel {
+            ip: ip,                       //tunnel endpoint
+            iface_name: iface_name,       //name of wg#
+            listen_ifidx: ifidx,          //The physical port this tunnel is listening on
+            listen_port: our_listen_port, //the port this tunnel resides on
+            //            tunnel_state: init,           //how this tunnel feels about it's life
+            localid: their_id.clone(), // the identity of the counterparty tunnel once we have it
+        };
```

Comment:
Thanks for commenting! Could you please move those to the struct *definition* above? That's where people curious about `Tunnel` will look.